src/lib.rs

```diff
@@ -1168,6 +1168,12 @@ impl Builder {
 
     /// Generate the Rust bindings using the options built up thus far.
     pub fn generate(mut self) -> Result<Bindings, ()> {
+        // Add any extra arguments from the environment to the clang command line.
+        if let Some(extra_clang_args) = std::env::var("BINDGEN_EXTRA_CLANG_ARGS").ok() {
+            self.options.clang_args.extend(extra_clang_args.split(" ").map(|s| s.to_string()));
```

Comment:
This will mishandle paths with spaces and such... We should probably use the `shlex` crate? Tough can be a followup if wanted I guess.